### PR TITLE
Move service config under stackdriver.metrics and simplify metric_types

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to move service config parameter from under `stackdriver` to under `stackdriver.metrics`. Service name used to only be used for checking metadata. Now it's also been used to construct the full `metric_type` string to make google cloud API call `list.timeseries`. 

For example, if `service=compute` and metric_type is given in config as `instance/cpu/usage_time`, then the full metric.type will be `compute/googleapi.com/instance/cpu/usage_time`.

This PR also removed the `stackdriver` config layer. Now instead of `stackdriver.metrics`, configs are directly under `metrics`.

Before:
```
- module: googlecloud
  metricsets:
    - stackdriver
  project_id: "your project id"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 1m
  stackdriver:
    service: compute
    metrics:
      - aligner: ALIGN_NONE
        metric_types:
          - "compute.googleapis.com/instance/cpu/reserved_cores"
          - "compute.googleapis.com/instance/cpu/usage_time"
          - "compute.googleapis.com/instance/cpu/utilization"
          - "compute.googleapis.com/instance/uptime"
```

After:
```
- module: googlecloud
  metricsets:
    - stackdriver
  project_id: "your project id"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 1m
  metrics:
    - aligner: ALIGN_NONE
      service: compute
      metric_types:
        - "instance/cpu/reserved_cores"
        - "instance/cpu/usage_time"
        - "instance/cpu/utilization"
        - "instance/uptime"
```

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
1. Enable googlecloud metricset: `./metricbeat modules enable googlecloud`
2. Change `modules.d/googlecloud.yml` to below:
```
- module: googlecloud
  metricsets:
    - stackdriver
  project_id: elastic-observability
  credentials_file_path: "/Users/kaiyansheng/Downloads/elastic-observability-d17781618202.json"
  exclude_labels: false
  period: 1m
  metrics:
    - aligner: ALIGN_NONE
      service: compute
      metric_types:
        - "instance/cpu/reserved_cores"
        - "instance/cpu/usage_time"
        - "instance/cpu/utilization"
        - "instance/uptime"
```
3. Metrics `googlecloud.stackdriver.instance.cpu.reserved_cores.value`, `googlecloud.stackdriver.instance.cpu.usage_time.value`, `googlecloud.stackdriver.instance.cpu.utilization.value` and `googlecloud.stackdriver.instance.uptime.value` should be collected.

### CODE REVIEW

File: x-pack/metricbeat/module/googlecloud/stackdriver/integration.go [hunk 1]
```diff
@@ -35,10 +35,10 @@ func GetConfigForTest(t *testing.T, metricSetName string) map[string]interface{}
 		}
 
 		if metricSetName == "stackdriver" {
-			config["stackdriver.service"] = "compute"
 			stackDriverConfig := stackDriverConfig{
+				ServiceName: "compute",
```
Comment (jsoriano) [inline x-pack/metricbeat/module/googlecloud/stackdriver/integration.go:39]: Nit. Use `googlecloud.ServiceCompute`?
```suggestion
				ServiceName: googlecloud.ServiceCompute,
```

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -431,6 +431,7 @@ field. You can revert this change by configuring tags for the module and omittin
 - Update MSSQL module to fix some SSPI authentication and add brackets to USE statements {pull}17862[17862]]
 - Add client address to events from http server module {pull}18336[18336]
 - Remove required for region/zone and make stackdriver a metricset in googlecloud. {issue}16785[16785] {pull}18398[18398]
+- Move service config under stackdriver.metrics and simplify metric types. {pull}18691[18691]
```
Comment (jsoriano) [inline CHANGELOG.next.asciidoc:434]: I think this is a breaking change.

File: x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go [hunk 3]
```diff
@@ -212,13 +219,15 @@ func (m *MetricSet) metricDescriptor(ctx context.Context, client *monitoring.Met
 		for _, mt := range sdc.MetricTypes {
 			req := &monitoringpb.ListMetricDescriptorsRequest{
 				Name:   "projects/" + m.config.ProjectID,
-				Filter: fmt.Sprintf(`metric.type = "%s"`, mt),
+				Filter: fmt.Sprintf(`metric.type = "%s"`, sdc.ServiceName+".googleapis.com/"+mt),
```
Comment (jsoriano) [inline x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go:222]: Is this correct for all services? Can the path be constructed the same way from the service name?

Also, can it happen that a service has metrics with different paths?

Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go:224]: I believe this is correct for all services. From the [google cloud documentation](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-compute), it shows under each service, `The following metric type names must be prefixed with xxx.googleapis.com/. That prefix has been omitted from the entries in the table.`

File: CHANGELOG.next.asciidoc (hunk 2 above)
Comment (kaiyan-sheng) [reply CHANGELOG.next.asciidoc:457]: @jsoriano But stackdriver metricset is in beta, should I still put the changelog under breaking change? 😬

File: x-pack/metricbeat/module/googlecloud/stackdriver/integration.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/googlecloud/stackdriver/integration.go:39]: Sorry why we need to change to `googlecloud.ServiceCompute`? I think this function is only for constructing config for testing purpose 😄 and the ServiceName here is equivalent to the `service` under `metrics`:
```
- module: googlecloud
  metricsets:
    - stackdriver
  project_id: "your project id"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 1m
  metrics:
    - aligner: ALIGN_NONE
      service: compute
      metric_types:
        - "instance/cpu/reserved_cores"
        - "instance/cpu/usage_time"
        - "instance/cpu/utilization"
        - "instance/uptime"
```

File: x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go (hunk 3 above)
Comment (jsoriano) [reply x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go:225]: Sounds good then :+1:

File: x-pack/metricbeat/module/googlecloud/stackdriver/integration.go (hunk 1 above)
Comment (jsoriano) [reply x-pack/metricbeat/module/googlecloud/stackdriver/integration.go:39]: I meant to use an existing constant, but you are right, we are creating a config for a test, no need to use a constant.

File: CHANGELOG.next.asciidoc (hunk 2 above)
Comment (jsoriano) [reply CHANGELOG.next.asciidoc]: I would say so, yes, even if it is beta there can be people using it.

Comment (kaiyan-sheng) [reply CHANGELOG.next.asciidoc]: @jsoriano Done! Thank you for the review!!

File: CHANGELOG.next.asciidoc [hunk 4]
```diff
@@ -54,7 +54,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 *Metricbeat*
 
 - Make use of secure port when accessing Kubelet API {pull}16063[16063]
-- Add Tomcat overview dashboard {pull}14026[14026]
```
Comment (kaiyan-sheng) [inline CHANGELOG.next.asciidoc:57]: Not sure why this is under breaking change, moving it to added section.

**Pull request merged**